pool/pool_manager.gno

```diff
@@ -220,46 +216,91 @@ func DoesPoolPathExist(poolPath string) bool {
 	return exist
 }
 
-// GetPool retrieves the pool for the given token paths and fee.
-// It constructs the poolPath from the given parameters and returns the corresponding pool.
-// Returns pool struct
+// GetPool retrieves a pool instance based on the provided token paths and fee tier.
+//
+// This function determines the pool path by combining the paths of token0 and token1 along with the fee tier,
+// and then retrieves the corresponding pool instance using that path.
+//
+// Parameters:
+// - token0Path (string): The unique identifier or path for token0.
+// - token1Path (string): The unique identifier or path for token1.
+// - fee (uint32): The fee tier for the pool, expressed in basis points (e.g., 3000 for 0.3%).
+//
+// Returns:
+// - *Pool: A pointer to the Pool instance corresponding to the provided tokens and fee tier.
+//
+// Notes:
+// - The order of token paths (token0Path and token1Path) matters and should match the pool's configuration.
+// - Ensure that the tokens and fee tier provided are valid and registered in the system.
+//
+// Example:
+// pool := GetPool("path/to/token0", "path/to/token1", 3000)
 func GetPool(token0Path, token1Path string, fee uint32) *Pool {
 	poolPath := GetPoolPath(token0Path, token1Path, fee)
-	pool, exist := pools[poolPath]
-	if !exist {
-		panic(addDetailToError(
-			errDataNotFound,
-			ufmt.Sprintf("pool_manager.gno__GetPool() || expected poolPath(%s) to exist", poolPath),
-		))
-	}
-
-	return pool
+	return GetPoolFromPoolPath(poolPath)
 }
 
-// GetPoolFromPoolPath retrieves the pool for the given poolPath.
+// GetPoolFromPoolPath retrieves a pool instance based on the provided pool path.
+//
+// This function checks if a pool exists for the given poolPath in the `pools` mapping.
+// If the pool exists, it returns the pool instance. Otherwise, it panics with a descriptive error.
+//
+// Parameters:
+// - poolPath (string): The unique identifier or path for the pool.
+//
+// Returns:
+// - *Pool: A pointer to the Pool instance corresponding to the given poolPath.
+//
+// Panics:
+//   - If the `poolPath` does not exist in the `pools` mapping, it panics with an error message
+//     indicating that the expected poolPath was not found.
+//
+// Notes:
+// - Ensure that the `poolPath` provided is valid and corresponds to an existing pool in the `pools` mapping.
+//
+// Example:
+// pool := GetPoolFromPoolPath("path/to/pool")
```

Comment:
Better to be actual pool path formula string

Reply:
@r3v4s I am concerned that if I provide the actual pool path, there will be hacking attempts utilizing this information.

Reply:
poolPath can be easily found on earn section of gnoswap web page. Therefore I don't think there are reason to hide it.

@notJoon your thoughts?

Reply:
It's just sample code, So I think any string would work as long as the format matches.